package/MDAnalysis/coordinates/chain.py

```diff
@@ -159,7 +158,7 @@ def check_allowed_filetypes(readers, allowed):
                                   "supported for formats: {}".format(allowed))
 
 
-class ChainReader(base.ProtoReader):
+class ChainReader(base.ReaderBase):
```

Comment:
this line change is what's happening in this PR.  Previously by subclassing ProtoReader (which is a more distant ancestor) you had to also reimplement how transformations worked.  

Instead ChainReader now subclasses `ReaderBase` (like a "normal" reader) and now only implements `_read_next_timestep` (like a "normal" Reader).  This means that methods `__next__`, `copy`, `add_transformations` are all done by the base class.

tl;dr less code = less bugs